File: aexample/src/main/java/org/judal/examples/java/jdbc/E30_GetNativeConnection.java
package org.judal.examples.java.jdbc;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;

import org.junit.Test;

import org.judal.storage.EngineFactory;
import org.judal.jdbc.JDBCRelationalDataSource;

public class E30_GetNativeConnection {

	@Test
	public void demo() throws Exception {
		
		setUp();

		JDBCRelationalDataSource dts = (JDBCRelationalDataSource) EngineFactory.getDefaultRelationalDataSource();

		Connection conn = dts.getConnection();		
		Statement stmt = conn.createStatement();
		ResultSet rset = stmt.executeQuery("SELECT * FROM student");
		
		// ...

		rset.close();
		stmt.close();
		conn.close();

		tearDown();
	}

	public static void setUp() throws Exception {
		E10_WriteCSVDataIntoTheDatabase.setUp();
	}

	public static void tearDown() throws Exception {
		E10_WriteCSVDataIntoTheDatabase.tearDown();
	}	

}

File: aexample/src/main/scala/org/judal/examples/scala/jdbc/E30_GetNativeConnection.scala
package org.judal.examples.scala.jdbc

import java.sql.Connection
import java.sql.ResultSet
import java.sql.Statement

import org.junit.Test

import org.judal.storage.EngineFactory
import org.judal.jdbc.JDBCRelationalDataSource

class E30_GetNativeConnection {

	@Test
	def demo() : Unit = {
		
		E30_GetNativeConnection.setUp()

		val dts = EngineFactory.getDefaultRelationalDataSource.asInstanceOf[JDBCRelationalDataSource]

		val conn = dts.getConnection
		val stmt = conn.createStatement
		val rset = stmt.executeQuery("SELECT * FROM student")

		// ...

		rset.close
		stmt.close
		conn.close

		E30_GetNativeConnection.tearDown()
	}

}


object E30_GetNativeConnection {
  
	def setUp() = E10_WriteCSVDataIntoTheDatabase.setUp()

	def tearDown() = E10_WriteCSVDataIntoTheDatabase.tearDown()

}